exercises/java/src/main/java/com/adaptionsoft/games/uglytrivia/Game.java

```diff
@@ -4,166 +4,193 @@
+        playersInPenaltyBox[getNumberOfPlayers()] = false;
+
+        System.out.println(playerName + " was added");
+        System.out.println("They are player number " + players.size());
+        return true;
+    }
+
+    private int getNumberOfPlayers() {
+        return players.size();
+    }
+
+    public void processRoll(int rolledNumber) {
+        System.out.println(players.get(currentPlayerID) + " is the current player");
+        System.out.println("They have rolled a " + rolledNumber);
+
+        if (isPlayerInPenaltyBox(currentPlayerID)) {
+            if (isOddNumber(rolledNumber)) {
+                isGettingOutOfPenaltyBox = true;
+
+                System.out.println(players.get(currentPlayerID) + " is getting out of the penalty box");
+                movePlayer(rolledNumber, currentPlayerID);
+
+                System.out.println("The category is " + getCategoryForPosition());
+                askQuestion();
+            } else {
+                System.out.println(players.get(currentPlayerID) + " is not getting out of the penalty box");
+                isGettingOutOfPenaltyBox = false;
+            }
+
+        } else {
+            movePlayer(rolledNumber, currentPlayerID);
+            System.out.println("The category is " + getCategoryForPosition());
+            askQuestion();
+        }
+
+    }
+
+    private void printPlayerPosition(int playerID) {
+        System.out.println(players.get(playerID)
+                + "'s new location is "
+                + getPlayerPosition(playerID));
+    }
+
+    private boolean isOddNumber(int number) {
+        return number % 2 != 0;
+    }
+
+    private void movePlayer(int rolledNumber, int playerID) {
+        playerPositions[playerID] = getPlayerPosition(playerID) + rolledNumber;
+        if (hasPlayerPassedFinalTile(playerID)) {
+            continueMoveFromStartOfBoard();
+        }
+
+        printPlayerPosition(playerID);
+    }
+
+    private void continueMoveFromStartOfBoard() {
+        playerPositions[currentPlayerID] = getPlayerPosition(currentPlayerID) - 12;
+    }
+
+    private boolean hasPlayerPassedFinalTile(int playerID) {
+        return getPlayerPosition(playerID) > 11;
+    }
+
+    private void askQuestion() {
+        if (getCategoryForPosition().equals(POP))
+            System.out.println(popQuestions.removeFirst());
+        if (getCategoryForPosition().equals(SCIENCE))
+            System.out.println(scienceQuestions.removeFirst());
+        if (getCategoryForPosition().equals(SPORTS))
+            System.out.println(sportsQuestions.removeFirst());
+        if (getCategoryForPosition().equals(ROCK))
+            System.out.println(rockQuestions.removeFirst());
+    }
+
+    private String getCategoryForPosition() {
+        return board[getPlayerPosition(currentPlayerID)];
+    }
+
+    private int getPlayerPosition(int playerID) {
+        return playerPositions[playerID];
+    }
+
+    public boolean handleCorrectAnswer() {

```

Comment:
This could perhaps be "playerAnsweredCorrectly" to keep in the context of the game, the same goes for handleIncorrectAnswer